src/test/java/teammates/test/cases/browsertests/AdminHomePageUiTest.java

```diff
@@ -83,15 +85,20 @@ private void testCreateInstructorAction() throws Exception {
         InstructorAttributes instructorInBackend = getInstructorWithRetry(demoCourseId, instructor.email);
         assertEquals("Instructor Details must have 3 columns", homePage.getMessageFromResultTable(1));
 
-        String encryptedKey = getKeyForInstructorWithRetry(demoCourseId, instructor.email);
-        // use AppUrl from Config because the join link takes its base URL from build.properties
-        String expectedjoinUrl = Config.getAppUrl(Const.ActionURIs.INSTRUCTOR_COURSE_JOIN)
-                                        .withRegistrationKey(encryptedKey)
-                                        .withInstructorInstitution(institute)
-                                        .toAbsoluteString();
         assertEquals("Instructor AHPUiT Instrúctör WithPlusInEmail has been successfully created " + Const.JOIN_LINK,
                 homePage.getMessageFromResultTable(2));
-        assertEquals(expectedjoinUrl, homePage.getJoinLink(homePage.getMessageFromResultTable(2)));
+
+        String encryptedKey = getKeyForInstructorWithRetry(demoCourseId, instructor.email);
+
+        String expectedjoinUrl = Config.getAppUrl(Const.ActionURIs.INSTRUCTOR_COURSE_JOIN)
+                                                        .withRegistrationKey(encryptedKey)
+                                                        .withInstructorInstitution(institute)
+                                                        .toAbsoluteString();
+        String expectedJoinLinkUrlFileName = new URL(expectedjoinUrl).getFile();
```

Comment:
Not really familiar; what does `URL#getFile()` return?

Reply:
Basically it returns getPath()+getQuery().

So for `https://teammates-john.appspot.com/page/adminPage?user=myself`, getFile() returns `/page/adminPage?user=myself`